mslice/cli/projection_functions.py

```diff
@@ -1,46 +1,108 @@
+from mslice.util.mantid.mantid_algorithms import Transpose
+from mslice.models.workspacemanager.workspace_algorithms import get_comment
+from mslice.models.labels import get_display_name, CUT_INTENSITY_LABEL
+from mantid.plots import plotfunctions
+from mslice.views.slice_plotter import create_slice_figure
+from mslice.views.slice_plotter import PICKER_TOL_PTS as SLICE_PICKER_TOL_PTS
+from mslice.views.cut_plotter import PICKER_TOL_PTS as CUT_PICKER_TOL_PTS
+from mslice.plotting.globalfiguremanager import GlobalFigureManager
 
 
-def PlotCutMsliceProjection(InputWorkspace, IntensityStart=0, IntensityEnd=0, PlotOver=False):
+@plt.set_category(plt.CATEGORY_CUT)
+def PlotCutMsliceProjection(axes, workspace, *args, **kwargs):
     """
     Same as the cli PlotCut but returns the relevant axes object.
     """
+    from mslice.app.presenters import get_cut_plotter_presenter
+    cur_fig = plt.gcf()
+    cur_canvas = cur_fig.canvas
 
-    _check_workspace_name(InputWorkspace)
-    workspace = get_workspace_handle(InputWorkspace)
+    _check_workspace_name(workspace)
+    workspace = get_workspace_handle(workspace)
     if not isinstance(workspace, HistogramWorkspace):
         raise RuntimeError("Incorrect workspace type.")
-    if IntensityStart == 0 and IntensityEnd == 0:
-        intensity_range = None
-    else:
-        intensity_range = (IntensityStart, IntensityEnd)
-    lines = cli_cut_plotter_presenter.plot_cut_from_workspace(workspace, intensity_range=intensity_range,
-                                                              plot_over=PlotOver)
 
-    return lines
+    presenter = get_cut_plotter_presenter()
+
+    plot_over = kwargs.pop('plot_over', True)
+    intensity_range = kwargs.pop('intensity_range', (None, None))
+    x_units = kwargs.pop('x_units', 'None')
+    legend = kwargs.pop('label', None)
+    legend = workspace.name if legend is None else legend
+
+    plotfunctions.errorbar(axes, workspace.raw_ws, label=legend, *args, **kwargs)
 
+    axes.set_ylim(*intensity_range) if intensity_range is not None else axes.autoscale()
+    if cur_canvas.manager.window.action_toggle_legends.isChecked():
+        leg = axes.legend(fontsize='medium')
+        leg.draggable()
+    axes.set_xlabel(get_display_name(x_units, get_comment(workspace)), picker=CUT_PICKER_TOL_PTS)
+    axes.set_ylabel(CUT_INTENSITY_LABEL, picker=CUT_PICKER_TOL_PTS)
+    if not plot_over:
+        cur_canvas.set_window_title(workspace.name)
+        cur_canvas.manager.update_grid()
+    if not cur_canvas.manager.has_plot_handler():
+        plot_handler = cur_canvas.manager.add_cut_plot(presenter, workspace.name.rsplit('_', 1)[0])
+        plot_handler.save_default_options()
+    cur_fig.canvas.draw()
 
-def PlotSliceMsliceProjection(InputWorkspace, IntensityStart="", IntensityEnd="", Colormap=DEFAULT_CMAP):
+    return axes.lines
+
+
+@plt.set_category(plt.CATEGORY_SLICE)
+def PlotSliceMsliceProjection(axes, workspace, *args, **kwargs):
```

Comment:
These function names don't really follow are naming conventions. They're not commands themselves so I think something like `errorbar` and `pcolormesh` would be fine names for these in a similar way that they are named in `mantid.plots.plotfunctions`. PyCharm rename should be able to help with this.